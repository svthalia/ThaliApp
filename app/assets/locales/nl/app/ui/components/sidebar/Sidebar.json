{
  "Log out?": "Uitloggen?",
  "Are you sure you want to log out?": "Weet je zeker dat je wilt uitloggen?",
  "No": "Nee",
  "Yes": "Ja",
  "Welcome": "Welkom",
  "Calendar": "Agenda",
  "Logout": "Uitloggen",
  "Member List": "Ledenlijst",
  "Settings": "Instellingen",
  "Photos": "Foto's"
}
