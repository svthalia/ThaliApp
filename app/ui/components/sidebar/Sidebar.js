import React from 'react';
import PropTypes from 'prop-types';
import {
  Alert, Image, ImageBackground, Text, TouchableHighlight, View,
} from 'react-native';
import LinearGradient from 'react-native-linear-gradient';
import { withTranslation } from 'react-i18next';
import Icon from 'react-native-vector-icons/MaterialIcons';
import styles from './style/Sidebar';
import Colors from '../../style/Colors';

const background = require('../../../assets/img/huygens.jpg');

const logoutPrompt = (props) => () => Alert.alert(
  props.t('Log out?'),
  props.t('Are you sure you want to log out?'),
  [{ text: props.t('No') },
    { text: props.t('Yes'), onPress: props.signOut },
  ],
);

const Sidebar = (props) => {
  const buttons = [
    {
      onPress: () => props.openWelcome(),
      iconName: 'home',
      text: props.t('Welcome'),
      style: {},
      routeName: 'Welcome',
    },
    {
      onPress: () => props.openCalendar(),
      iconName: 'event',
      text: props.t('Calendar'),
      style: {},
      routeName: 'Calendar',
    },
    {
      onPress: () => props.openMemberList(),
      iconName: 'people',
      text: props.t('Member List'),
      style: {},
      routeName: 'MemberList',
    },
    {
      onPress: () => props.openPhotos(),
      iconName: 'photo',
      text: props.t('Photos'),
      style: {},
      routeName: 'Photos',
    },
    {
      onPress: () => props.openSettings(),
      iconName: 'settings',
      text: props.t('Settings'),
      style: {},
      routeName: 'Settings',
    },
    {
      onPress: logoutPrompt(props),
      iconName: 'lock',
      text: props.t('Logout'),
      style: {
        borderTopColor: Colors.lightGray,
        borderTopWidth: 1,
      },
      routeName: 'signOut',
    },
  ];

  return (
    <View
      style={styles.sidebar}
    >
      <TouchableHighlight
        onPress={() => props.loadProfile()}
        style={styles.headerButton}
      >
        <ImageBackground
          source={background}
          style={styles.headerImage}
          resizeMode="cover"
        >
          <LinearGradient colors={['#55000000', '#000000']} style={styles.headerGradient} />
          <Image
            source={{ uri: props.photo }}
            style={styles.profileImage}
            resizeMode="cover"
          />
          <Text style={styles.nameField}>
            {props.displayName}
          </Text>
        </ImageBackground>
      </TouchableHighlight>
      <View style={styles.buttonList}>
        {buttons.map((button) => (
          <Icon.Button
            onPress={button.onPress}
            name={button.iconName}
            borderRadius={0}
            backgroundColor={Colors.white}
            color={props.activeItemKey === button.routeName
              ? Colors.magenta : Colors.textColour}
            size={24}
            iconStyle={styles.buttonIcon}
            style={[styles.button, button.style]}
            key={button.routeName}
          >
            <Text style={[styles.buttonText, {
              color: props.activeItemKey === button.routeName
                ? Colors.magenta : Colors.textColour,
            }]}
            >
              {button.text}
            </Text>
          </Icon.Button>
        ))}
      </View>
    </View>
  );
};

Sidebar.propTypes = {
  activeItemKey: PropTypes.string.isRequired,
  displayName: PropTypes.string.isRequired,
  photo: PropTypes.string.isRequired,
  loadProfile: PropTypes.func.isRequired,
  t: PropTypes.func.isRequired,
  openCalendar: PropTypes.func.isRequired,
  openWelcome: PropTypes.func.isRequired,
  openSettings: PropTypes.func.isRequired,
  openMemberList: PropTypes.func.isRequired,
  openPhotos: PropTypes.func.isRequired,
};

export default withTranslation('ui/components/sidebar/Sidebar')(Sidebar);
