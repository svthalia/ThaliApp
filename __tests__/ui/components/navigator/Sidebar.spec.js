import React from 'react';
import renderer from 'react-test-renderer';
import Sidebar from '../../../../app/ui/components/sidebar/Sidebar';

jest.mock('react-native-vector-icons/MaterialIcons', () => ({
  Button: 'IconButton',
}));

jest.mock('@react-native-community/status-bar', () => 'StatusBar');

describe('Sidebar component', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(<Sidebar
        activeItemKey="Welcome"
        photo="photo"
        displayName="Lorem ipsum"
        openCalendar={jest.fn()}
        openSettings={jest.fn()}
        openWelcome={jest.fn()}
        openMemberList={jest.fn()}
        loadProfile={jest.fn()}
      />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
