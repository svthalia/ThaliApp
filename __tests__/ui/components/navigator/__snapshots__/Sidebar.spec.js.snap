// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sidebar component renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "stretch",
      "backgroundColor": "#FFFFFF",
      "flex": 1,
    }
  }
>
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "height": 168,
      }
    }
  >
    <View
      accessibilityIgnoresInvertColors={true}
      style={
        Object {
          "flexDirection": "column",
          "height": 168,
          "justifyContent": "space-between",
          "padding": 16,
          "paddingTop": 36,
          "width": null,
        }
      }
    >
      <Image
        resizeMode="cover"
        source={
          Object {
            "testUri": "../../../app/assets/img/huygens.jpg",
          }
        }
        style={
          Array [
            Object {
              "bottom": 0,
              "left": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
            },
            Object {
              "height": 168,
              "width": null,
            },
            undefined,
          ]
        }
      />
      <BVLinearGradient
        colors={
          Array [
            5570560,
            4278190080,
          ]
        }
        endPoint={
          Object {
            "x": 0.5,
            "y": 1,
          }
        }
        locations={null}
        startPoint={
          Object {
            "x": 0.5,
            "y": 0,
          }
        }
        style={
          Object {
            "bottom": 0,
            "left": 0,
            "opacity": 0.6,
            "position": "absolute",
            "right": 0,
            "top": 0,
          }
        }
      />
      <Image
        resizeMode="cover"
        source={
          Object {
            "uri": "photo",
          }
        }
        style={
          Object {
            "borderRadius": 32,
            "height": 64,
            "width": 64,
          }
        }
      />
      <Text
        style={
          Object {
            "color": "#FFFFFF",
            "fontFamily": "System",
            "fontSize": 14,
            "fontWeight": "600",
          }
        }
      >
        Lorem ipsum
      </Text>
    </View>
  </View>
  <View
    style={
      Object {
        "flex": 3,
      }
    }
  >
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          Object {
            "overflow": "hidden",
          },
          Object {
            "backgroundColor": "#FFFFFF",
            "borderRadius": 0,
          },
        ]
      }
    >
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "flexDirection": "row",
              "justifyContent": "flex-start",
              "padding": 8,
            },
            Object {
              "backgroundColor": "#FFFFFF",
              "borderRadius": 0,
            },
            Array [
              Object {
                "padding": 16,
              },
              Object {},
            ],
          ]
        }
      >
        <Text
          allowFontScaling={false}
          style={
            Array [
              Object {
                "color": "#E62272",
                "fontSize": 24,
              },
              Array [
                Object {
                  "marginRight": 10,
                },
                Object {
                  "marginRight": 30,
                  "textAlign": "center",
                  "width": 28,
                },
              ],
              Object {
                "fontFamily": "Material Icons",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
              Object {},
            ]
          }
        >
          
        </Text>
        <Text
          style={
            Array [
              Object {
                "color": "#313131",
                "fontFamily": "System",
                "fontSize": 14,
                "fontWeight": "600",
              },
              Object {
                "color": "#E62272",
              },
            ]
          }
        >
          Welcome
        </Text>
      </View>
    </View>
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          Object {
            "overflow": "hidden",
          },
          Object {
            "backgroundColor": "#FFFFFF",
            "borderRadius": 0,
          },
        ]
      }
    >
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "flexDirection": "row",
              "justifyContent": "flex-start",
              "padding": 8,
            },
            Object {
              "backgroundColor": "#FFFFFF",
              "borderRadius": 0,
            },
            Array [
              Object {
                "padding": 16,
              },
              Object {},
            ],
          ]
        }
      >
        <Text
          allowFontScaling={false}
          style={
            Array [
              Object {
                "color": "#313131",
                "fontSize": 24,
              },
              Array [
                Object {
                  "marginRight": 10,
                },
                Object {
                  "marginRight": 30,
                  "textAlign": "center",
                  "width": 28,
                },
              ],
              Object {
                "fontFamily": "Material Icons",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
              Object {},
            ]
          }
        >
          
        </Text>
        <Text
          style={
            Array [
              Object {
                "color": "#313131",
                "fontFamily": "System",
                "fontSize": 14,
                "fontWeight": "600",
              },
              Object {
                "color": "#313131",
              },
            ]
          }
        >
          Calendar
        </Text>
      </View>
    </View>
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          Object {
            "overflow": "hidden",
          },
          Object {
            "backgroundColor": "#FFFFFF",
            "borderRadius": 0,
          },
        ]
      }
    >
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "flexDirection": "row",
              "justifyContent": "flex-start",
              "padding": 8,
            },
            Object {
              "backgroundColor": "#FFFFFF",
              "borderRadius": 0,
            },
            Array [
              Object {
                "padding": 16,
              },
              Object {},
            ],
          ]
        }
      >
        <Text
          allowFontScaling={false}
          style={
            Array [
              Object {
                "color": "#313131",
                "fontSize": 24,
              },
              Array [
                Object {
                  "marginRight": 10,
                },
                Object {
                  "marginRight": 30,
                  "textAlign": "center",
                  "width": 28,
                },
              ],
              Object {
                "fontFamily": "Material Icons",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
              Object {},
            ]
          }
        >
          
        </Text>
        <Text
          style={
            Array [
              Object {
                "color": "#313131",
                "fontFamily": "System",
                "fontSize": 14,
                "fontWeight": "600",
              },
              Object {
                "color": "#313131",
              },
            ]
          }
        >
          Member List
        </Text>
      </View>
    </View>
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          Object {
            "overflow": "hidden",
          },
          Object {
            "backgroundColor": "#FFFFFF",
            "borderRadius": 0,
          },
        ]
      }
    >
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "flexDirection": "row",
              "justifyContent": "flex-start",
              "padding": 8,
            },
            Object {
              "backgroundColor": "#FFFFFF",
              "borderRadius": 0,
            },
            Array [
              Object {
                "padding": 16,
              },
              Object {},
            ],
          ]
        }
      >
        <Text
          allowFontScaling={false}
          style={
            Array [
              Object {
                "color": "#313131",
                "fontSize": 24,
              },
              Array [
                Object {
                  "marginRight": 10,
                },
                Object {
                  "marginRight": 30,
                  "textAlign": "center",
                  "width": 28,
                },
              ],
              Object {
                "fontFamily": "Material Icons",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
              Object {},
            ]
          }
        >
          
        </Text>
        <Text
          style={
            Array [
              Object {
                "color": "#313131",
                "fontFamily": "System",
                "fontSize": 14,
                "fontWeight": "600",
              },
              Object {
                "color": "#313131",
              },
            ]
          }
        >
          Photos
        </Text>
      </View>
    </View>
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          Object {
            "overflow": "hidden",
          },
          Object {
            "backgroundColor": "#FFFFFF",
            "borderRadius": 0,
          },
        ]
      }
    >
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "flexDirection": "row",
              "justifyContent": "flex-start",
              "padding": 8,
            },
            Object {
              "backgroundColor": "#FFFFFF",
              "borderRadius": 0,
            },
            Array [
              Object {
                "padding": 16,
              },
              Object {},
            ],
          ]
        }
      >
        <Text
          allowFontScaling={false}
          style={
            Array [
              Object {
                "color": "#313131",
                "fontSize": 24,
              },
              Array [
                Object {
                  "marginRight": 10,
                },
                Object {
                  "marginRight": 30,
                  "textAlign": "center",
                  "width": 28,
                },
              ],
              Object {
                "fontFamily": "Material Icons",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
              Object {},
            ]
          }
        >
          
        </Text>
        <Text
          style={
            Array [
              Object {
                "color": "#313131",
                "fontFamily": "System",
                "fontSize": 14,
                "fontWeight": "600",
              },
              Object {
                "color": "#313131",
              },
            ]
          }
        >
          Settings
        </Text>
      </View>
    </View>
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          Object {
            "overflow": "hidden",
          },
          Object {
            "backgroundColor": "#FFFFFF",
            "borderRadius": 0,
          },
        ]
      }
    >
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "flexDirection": "row",
              "justifyContent": "flex-start",
              "padding": 8,
            },
            Object {
              "backgroundColor": "#FFFFFF",
              "borderRadius": 0,
            },
            Array [
              Object {
                "padding": 16,
              },
              Object {
                "borderTopColor": "#BBBBBB",
                "borderTopWidth": 1,
              },
            ],
          ]
        }
      >
        <Text
          allowFontScaling={false}
          style={
            Array [
              Object {
                "color": "#313131",
                "fontSize": 24,
              },
              Array [
                Object {
                  "marginRight": 10,
                },
                Object {
                  "marginRight": 30,
                  "textAlign": "center",
                  "width": 28,
                },
              ],
              Object {
                "fontFamily": "Material Icons",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
              Object {},
            ]
          }
        >
          
        </Text>
        <Text
          style={
            Array [
              Object {
                "color": "#313131",
                "fontFamily": "System",
                "fontSize": 14,
                "fontWeight": "600",
              },
              Object {
                "color": "#313131",
              },
            ]
          }
        >
          Logout
        </Text>
      </View>
    </View>
  </View>
</View>
`;
